loadbalancer/loadbalancer.go

```diff
@@ -0,0 +1,289 @@
+package loadbalancer
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"sync"
+	"time"
+
+	"github.com/filecoin-project/boost/loadbalancer/messages"
+	logging "github.com/ipfs/go-log/v2"
+	"github.com/libp2p/go-libp2p-core/host"
+	"github.com/libp2p/go-libp2p-core/network"
+	"github.com/libp2p/go-libp2p-core/peer"
+	"github.com/libp2p/go-libp2p-core/protocol"
+	"github.com/multiformats/go-multiaddr"
+)
+
+const DefaultDuration = time.Hour
+
+var log = logging.Logger("loadbalancer")
+
+type LoadBalancer struct {
```

Comment:
I agree with Rod's comment that LoadBalancer is somewhat of a misnomer. For example it doesn't make sense to load balance bitswap requests as bitswap is stateful.